### PR TITLE
feat: [kmsinventory] new module for kmsinventory

### PR DESCRIPTION
**Pull request by emmileaf**
Initial client library generation for `kmsinventory`:
```
python3.9 generation/new_client/new-client.py generate \
  --api_shortname=kmsinventory \
  --proto-path=google/cloud/kms/inventory \
  --name-pretty="KMS Inventory API" \
  --product-docs="https://cloud.google.com/kms/docs/" \
  --api-description="KMS Inventory API."
```

Note: there is no DevRel Services product page for this API yet (hence the failing snippetbot checks), and parameters above were collected from ticket (pointing to https://github.com/googleapis/googleapis/blob/master/google/cloud/kms/inventory/v1/kmsinventory_v1.yaml).

Summary of manual adjustments:
- Added dependency on `proto-google-cloud-kms-v1`: [98daccc](https://github.com/googleapis/google-cloud-java/pull/9162/commits/98daccc4dafa2991ab173f8f7ab80adf695ce805) and [5fe62af](https://github.com/googleapis/google-cloud-java/pull/9162/commits/5fe62af316b1517b59308b53fdaff429b20933d6)
- Added version update annotation to readme: [afa8738](https://github.com/googleapis/google-cloud-java/pull/9162/commits/afa8738719c29695f2dba266fbea0c3d52d8f0d1)

### CODE REVIEW

File: java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponseOrBuilder.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  /**
+   *
+   *
+   * <pre>
+   * The list of [CryptoKeys][google.cloud.kms.v1.CryptoKey].
+   * </pre>
+   *
+   * <code>repeated .google.cloud.kms.v1.CryptoKey crypto_keys = 1;</code>
+   */
+  java.util.List<com.google.cloud.kms.v1.CryptoKey> getCryptoKeysList();
```
Comment (emmileaf) [inline java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponseOrBuilder.java:35]: Compilation error ([logs](https://github.com/googleapis/google-cloud-java/actions/runs/4284102518/jobs/7460588009)), from dependency on `com.google.cloud.kms.v1`
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project proto-google-cloud-kmsinventory-v1: Compilation failure: Compilation failure: 
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponseOrBuilder.java:[35,41] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponseOrBuilder.java:[45,26] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponseOrBuilder.java:[65,51] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponseOrBuilder.java:[75,26] package com.google.cloud.kms.v1 does not exist
```

File: java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java [hunk 2]
```diff
@@ -0,0 +1,1131 @@
...
+        .internal_static_google_cloud_kms_inventory_v1_ListCryptoKeysResponse_fieldAccessorTable
+        .ensureFieldAccessorsInitialized(
+            com.google.cloud.kms.inventory.v1.ListCryptoKeysResponse.class,
+            com.google.cloud.kms.inventory.v1.ListCryptoKeysResponse.Builder.class);
+  }
+
+  public static final int CRYPTO_KEYS_FIELD_NUMBER = 1;
+
+  @SuppressWarnings("serial")
+  private java.util.List<com.google.cloud.kms.v1.CryptoKey> cryptoKeys_;
```
Comment (emmileaf) [inline java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:75]: Compilation error ([logs](https://github.com/googleapis/google-cloud-java/actions/runs/4284102518/jobs/7460588009)), from dependency on `com.google.cloud.kms.v1`
```
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[75,49] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[86,48] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[99,58] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[126,33] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[139,33] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[612,51] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[623,36] package com.google.cloud.kms.v1 does not exist
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-kmsinventory/proto-google-cloud-kmsinventory-v1/src/main/java/com/google/cloud/kms/inventory/v1/ListCryptoKeysResponse.java:[624,46] package com.google.cloud.kms.v1.CryptoKey does not exist
...
```

File: java-kmsinventory/pom.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      <dependency>
+        <groupId>com.google.api.grpc</groupId>
+        <artifactId>grpc-google-cloud-kmsinventory-v1</artifactId>
+        <version>0.0.1-SNAPSHOT</version><!-- {x-version-update:grpc-google-cloud-kmsinventory-v1:current} -->
+      </dependency>
+      <dependency>
+        <groupId>com.google.api.grpc</groupId>
+        <artifactId>proto-google-cloud-kmsinventory-v1</artifactId>
+        <version>0.0.1-SNAPSHOT</version><!-- {x-version-update:proto-google-cloud-kmsinventory-v1:current} -->
+      </dependency>
```
Comment (blakeli0) [inline java-kmsinventory/pom.xml:43]: We probably need to add a section like this to here:
```
  <groupId>com.google.api.grpc</groupId>
  <artifactId>proto-google-cloud-kms-v1</artifactId>
  <version>0.106.0-SNAPSHOT</version><!-- {x-version-update:proto-google-cloud-kms-v1:current} -->
```
to fix the compilation issue. 
@suztomo @meltsufin FYI, this is a rare case that an API is dependent on another API's proto, I think it should be fine as long as the versions are centrally maintained in versions.txt, but want to make sure this is not a problem for releasing.

**Comment by lqiu96** [comment]
I believe there was an older thread regarding this library using cross-api dependencies: https://chat.google.com/room/AAAAYlCaXjg/sqzmXYYnbX8

Blake mentioned that asset already does this and maybe we can do something similar: https://github.com/googleapis/google-cloud-java/blob/5fd6f610f52824b4a25b0b455a4d9503c0de72b8/java-asset/pom.xml#L84-L93

**Comment by emmileaf** [comment]
@blakeli0 @lqiu96 Ah sorry I just saw your comments after pushing [f181a73](https://github.com/googleapis/google-cloud-java/pull/9162/commits/f181a734dbbfa11ee3335d191fca5f297b8a05b1)! Would adding a dependency on `proto-google-cloud-kms-v1` be preferred here to adding `google-cloud-kms` as a manual dependency adjustment?

**Comment by lqiu96** [comment]
> @blakeli0 @lqiu96 Ah sorry I just saw your comments after pushing [f181a73](https://github.com/googleapis/google-cloud-java/pull/9162/commits/f181a734dbbfa11ee3335d191fca5f297b8a05b1)! Would adding a dependency on `proto-google-cloud-kms-v1` be preferred here to adding `google-cloud-kms` as a manual dependency adjustment?

I think adding the proto module dependency is be preferred (though not sure what everyone else's opinion is). Looks like we're only dependent on `CryptoKey` in the `proto-*` module (https://github.com/googleapis/google-cloud-java/blob/5fd6f610f52824b4a25b0b455a4d9503c0de72b8/java-kms/proto-google-cloud-kms-v1/src/main/java/com/google/cloud/kms/v1/CryptoKey.java).

**Comment by emmileaf** [comment]
The checks are hitting the same compilation errors - locally running `mvn -f java-kmsinventory compile` and `mvn compile` succeeds, but running the exact command as `build.sh` fails:

https://github.com/googleapis/google-cloud-java/blob/5fd6f610f52824b4a25b0b455a4d9503c0de72b8/.kokoro/build.sh#L41-L49   

Perhaps there's something else needed (for test scope)? Going to experiment a bit more locally.

**Comment by emmileaf** [comment]
Update: ci/unit checks are passing now - thanks for the help @blakeli0 and @lqiu96 !

Manual dependency additions made in [98daccc](https://github.com/googleapis/google-cloud-java/pull/9162/commits/98daccc4dafa2991ab173f8f7ab80adf695ce805) and [5fe62af](https://github.com/googleapis/google-cloud-java/pull/9162/commits/5fe62af316b1517b59308b53fdaff429b20933d6):

https://github.com/googleapis/google-cloud-java/blob/5fe62af316b1517b59308b53fdaff429b20933d6/java-kmsinventory/proto-google-cloud-kmsinventory-v1/pom.xml#L36-L39

https://github.com/googleapis/google-cloud-java/blob/5fe62af316b1517b59308b53fdaff429b20933d6/java-kmsinventory/pom.xml#L44-L49

**Review by lqiu96 — APPROVED** [review]

File: java-kmsinventory/README.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+
+If you are using Maven, add this to your pom.xml file:
+
+
+```xml
+<dependency>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>google-cloud-kmsinventory</artifactId>
+  <version>0.0.0</version>
```
Comment (blakeli0) [inline java-kmsinventory/README.md:25]: We need to manually add release please set up for this version

Comment (emmileaf) [reply java-kmsinventory/README.md:26]: Added in [afa8738](https://github.com/googleapis/google-cloud-java/pull/9162/commits/afa8738719c29695f2dba266fbea0c3d52d8f0d1), following https://github.com/googleapis/google-cloud-java/pull/9150 (cc/ @ddixit14)

Comment (blakeli0) [reply java-kmsinventory/README.md:26]: Thanks! Can you please add it to our new module creation guide as well? cc @suztomo

**Review by ddixit14 — APPROVED** [review]

**Pull request merged**